proxy-node-shared/src/main/java/uk/gov/ida/notification/validations/DestinationUriStringValidator.java

```diff
@@ -0,0 +1,33 @@
+package uk.gov.ida.notification.validations;
+
+import org.apache.commons.lang3.StringUtils;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.net.URI;
+
+public class DestinationUriStringValidator implements ConstraintValidator<ValidDestinationUriString, String> {
+
+    @Override
+    public void initialize(ValidDestinationUriString constraint) { /* intentionally blank */ }
+
+    @Override
+    public boolean isValid(String value, ConstraintValidatorContext context) {
+        if (StringUtils.isBlank(value)) { return true; } // @NotNull should detect nulls
+
+        try {
+            URI uri = URI.create(value);
+            if (uri.getScheme().equalsIgnoreCase("http") ||
+                uri.getScheme().equalsIgnoreCase("https")) {
+                return true;
+            } else {
```

Comment:
wrapping in else is redundant

Reply:
It felt clearer to me. I'm happy to remove the wrap.